packages/selection/src/useSelection.js

```diff
@@ -208,6 +213,7 @@ export function useSelection({
       role,
       tabIndex,
       [selectedAriaKey]: isSelected,
+      ref: focusRef,
```

Comment:
There will be some issues with us assuming that all elements that this prop-getter is spread on have a valid `ref` attribute.  For example, the majority of our existing Garden components still rely on the `innerRef` prop.  The separation previously was to allow this.

Also, how does this work if there is a need to share the `ref` with multiple prop-getters? This would wipe-out other uses I think.

Reply:
> Also, how does this work if there is a need to share the ref with multiple prop-getters? This would wipe-out other uses I think.

Can you elaborate on that a bit more I'm not sure I understand what you mean

Reply:
Lets say you have a component that needs to apply the `useSelection` and (just as an example) `useTooltip`. If both of these hooks require access to the same ref object, I don't think it would be shared with this change.

Example:
```jsx
const { getItemProps } = useSelection();
const { getTriggerProps } = useTooltip();

// After change - how does this work if the `ref` object is actually passed through?
<div {...getItemProps(getTriggerProps())}>hello</div>

const divRef = useRef(null);

// Before change - looks gross, but this could be composed if necessary
<div {...getItemProps({ 
        refKey: divRef, 
        ...getTriggerProps({ refKey: divRef, ref: divRef })
    })}>
    hello
</div>
```

This also fixes the issue with some items not using the `ref` key

Reply:
This works with the revised `useSelection` https://codesandbox.io/s/kmmn1234xr

Reply:
Here is an example of what I'm thinking of https://codesandbox.io/s/ymy3qrn2rx In the current implementation the direct component _must_ have a `ref` key that is handled correctly. That isn't (normally) always the case.

Reply:
So in your example you could just add `innerRef: itemRefs[index]` in the `getItemProps` setter? It's still a single ref to the underlying dom node.

Reply:
Let's schedule some face-to-face time to go through this some 😄 Ha I'm confused with this

Reply:
From our offline conversation let's add in an optional `[refKey="ref"]` attribute that can be used to allow alternate ref naming.

Otherwise the ability to compose and nest these with other hooks should be good! Thanks for helping me understand it.